### PR TITLE
Create Boxed(Value) newtype for Boxed GValue

### PR DESCRIPTION
**Pull request by mehmooda**
Part of #4

### CODE REVIEW

File: glib/src/value.rs [hunk 1]
```diff
@@ -1005,45 +1005,47 @@ numeric!(
     gobject_ffi::g_value_set_double
 );
 
-impl ValueType for Value {
-    type Type = Value;
+pub struct Boxed(pub Value);
```
Comment (sdroege) [inline glib/src/value.rs:1008]:
```suggestion
pub struct BoxedValue(pub Value);
```

And then also re-export it from `src/lib.rs`

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise, thanks!

File: glib/src/value.rs [hunk 2]
```diff
@@ -1005,45 +1005,47 @@ numeric!(
     gobject_ffi::g_value_set_double
 );
 
-impl ValueType for Value {
-    type Type = Value;
+pub struct BoxedValue(pub Value);
```
Comment (sdroege) [inline glib/src/value.rs:1008]: Needs some docs and having an `impl Deref for BoxedValue` would seem useful

**Comment by sdroege** [comment]
@mehmooda If you want it in this release then it has to be updated in the next few days.

File: glib/src/value.rs (hunk 2 above)
Comment (sdroege) [reply glib/src/value.rs:1014]: Added both

File: glib/src/value.rs [hunk 3]
```diff
@@ -1010,45 +1010,56 @@ numeric!(
     gobject_ffi::g_value_set_double
 );
 
-impl ValueType for Value {
-    type Type = Value;
+/// A [`Value`] containing another [`Value`].
+pub struct BoxedValue(pub Value);
```
[the remark this answers is not in the document]
Comment (sdroege) [reply glib/src/value.rs:1014]: Yes

**Comment by bilelmoussaoui** [comment]
I suppose this change doesn't require any change from gir's side? otherwise looks good to me

**Comment by sdroege** [comment]
> I suppose this change doesn't require any change from gir's side? otherwise looks good to me

They don't :)

**Pull request merged**